controller/src/beerocks/master/son_management.cpp

```diff
@@ -1803,26 +1803,355 @@ void son_management::handle_bml_message(Socket *sd,
+                          "cACTION_BML_CHANNEL_SCAN_SET_CONTINUOUS_ENABLE_RESPONSE !";
+            return;
+        }
+
+        auto radio_mac = request->radio_mac();
+        auto enable    = (request->isEnable() == 1);
+        LOG(DEBUG) << "request radio_mac:" << radio_mac;
+        bool success = database.set_channel_scan_is_enabled(radio_mac, enable);
+        response->op_error_code() =
+            uint8_t((success) ? eChannelScanOpErrCode::CHANNEL_SCAN_OP_SUCCESS
+                              : eChannelScanOpErrCode::CHANNEL_SCAN_OP_INVALID_PARAMS_ENABLE);
+
+        // on DCS enable - "reset" the interval wait of the task
+        // (will perform scan right after change to enable)
+        if (enable && success) {
+            dynamic_channel_selection_task::sScanEvent new_event;
+            new_event.radio_mac = request->radio_mac();
+
+            tasks.push_event(database.get_dynamic_channel_selection_task_id(radio_mac),
+                             (int)dynamic_channel_selection_task::eEvent::SCAN_ENABLE_CHANGE,
+                             (void *)&new_event);
+        }
+        //send response to bml
+        message_com::send_cmdu(sd, cmdu_tx);
         break;
     }
     case beerocks_message::ACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_REQUEST: {
         LOG(TRACE) << "ACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_REQUEST";
+
+        auto request = beerocks_header->addClass<
+            beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_REQUEST failed";
+            break;
+        }
+
+        auto response = message_com::create_vs_message<
+            beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_RESPONSE>(
+            cmdu_tx, beerocks_header->id());
+        if (!response) {
+            LOG(ERROR) << "Failed building message "
+                          "cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_ENABLE_RESPONSE !";
+            return;
+        }
+
+        auto radio_mac = request->radio_mac();
+        LOG(DEBUG) << "request radio_mac:" << radio_mac;
+        response->isEnable() = (database.get_channel_scan_is_enabled(radio_mac)) ? 1 : 0;
+
+        //send response to bml
+        message_com::send_cmdu(sd, cmdu_tx);
         break;
     }
     case beerocks_message::ACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST: {
         LOG(TRACE) << "ACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST";
+
+        auto request =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST failed";
+            break;
+        }
+
+        // Get information from request
+        auto radio_mac      = request->radio_mac();
+        auto is_single_scan = request->scan_mode() == 1;
```

Comment:
There's no constant or enum for the scan modes?
Are there any more modes or is it just 0 and 1?

Reply:
just 0 & 1 since we don't send booleans over CMDU

Reply:
So define an enum in the yaml and use it instead

Reply:
nitpick anyway..